### PR TITLE
[14.0] [IMP] purchase_sale_inter_company: sync pickings between companies

### PR DESCRIPTION
**Pull request by renda-dev**
Migration of https://github.com/OCA/multi-company/pull/362

Added some fixes and migration to v14

### Disclaimer:
I've added some things in the CI's tests matrix because this module can't run it's test with `stock_intercompany` installed, since the purposes of these two modules are quite similar.

Also, I can't exclude `stock_intercompany` in this module's `manifest.py` because the runboat would not even start.

I'm totally open to suggestion on how we can handle this situation.

### CODE REVIEW

**Comment by MiquelRForgeFlow** [comment]
@renda-dev Maybe you are interested in https://github.com/OCA/multi-company/pull/508.

**Comment by francesco-ooops** [comment]
@MiquelRForgeFlow would you mind reviewing this one?

We'll add tests soon.

**Comment by MiquelRForgeFlow** [comment]
> We'll add tests soon.

Then do you mind if I wait for the tests? haha

**Review by francesco-ooops — APPROVED** [review]
Functional ok!

**Comment by francesco-ooops** [comment]
@MiquelRForgeFlow @AaronHForgeFlow ready for review!

File: purchase_sale_inter_company/__manifest__.py [hunk 1]
```diff
@@ -12,6 +12,15 @@
     "author": "Odoo SA, Akretion, Tecnativa, Odoo Community Association (OCA)",
     "license": "AGPL-3",
     "installable": True,
-    "depends": ["sale", "purchase", "stock", "account_invoice_inter_company"],
-    "data": ["views/res_config_view.xml"],
+    "depends": [
+        "sale",
+        "sale_stock",
+        "purchase",
+        "stock",
```
Comment (MiquelRForgeFlow) [inline purchase_sale_inter_company/__manifest__.py]:
```suggestion
        "sale_stock",
        "purchase_stock",
```

**Review by ilyasProgrammer — APPROVED** [review]
Code LGTM

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow shall we merge? :)

**Comment by francesco-ooops** [comment]
@AaronHForgeFlow we got to cover with tests cases added by our changes, but not the ones from original PR. 

Do you think it's good for merge? 

Thank you!

**Comment by francesco-ooops** [comment]
@JordiBForgeFlow can this be merged? AFAICS you're the only active PSC in this repo

**Review by AaronHForgeFlow — APPROVED** [review]
code review LGTM

**Comment by francesco-ooops** [comment]
@pedrobaeza could you please merge based on existing reviews?

**Comment by pedrobaeza** [comment]
On later versions, the stock part is extracted on a separate module. I don't like either all the exclude thing, but I'm not going to block this:

/ocabot merge major

**Comment by francesco-ooops** [comment]
@pedrobaeza could you please relaunch?

**Comment by francesco-ooops** [comment]
@pedrobaeza the bots are on strike apparently

**Comment by MiquelRForgeFlow** [comment]
Maybe the PR needs a rebase.

**Comment by renda-dev** [comment]
Rebase done

**Comment by francesco-ooops** [comment]
@pedrobaeza try again please?

**Comment by francesco-ooops** [comment]
@pedrobaeza any idea on how to get out of this situation? It happened before but I can't remember what we did to fix it.

**Comment by pedrobaeza** [comment]
This has happened before if the setup folders are manually created instead of created by pre-commit.

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -43,10 +43,17 @@ jobs:
...
-            exclude: "product_tax_multicompany_default"
+            include: "stock_intercompany"
             name: test with Odoo
           - container: ghcr.io/oca/oca-ci/py3.6-ocb14.0:latest
-            exclude: "product_tax_multicompany_default"
+            include: "stock_intercompany"
+            name: test with OCB
+          - container: ghcr.io/oca/oca-ci/py3.6-odoo14.0:latest
+            exclude: "product_tax_multicompany_default,stock_intercompany"
+            makepot: "true"
```
Comment (MiquelRForgeFlow) [inline .github/workflows/test.yml]:
```suggestion
            makepot: "true"
          - container: ghcr.io/oca/oca-ci/py3.6-odoo14.0:latest
            exclude: "product_tax_multicompany_default,stock_intercompany"
```

**Review by MiquelRForgeFlow — COMMENTED** [review]
In case you want to do some change, in addition do this.

File: .github/workflows/test.yml (hunk 2 above)
Comment (renda-dev) [reply .github/workflows/test.yml]: It actually might be the solution, since it was generating recursively lots of commits with `.pot` files.

**Comment by renda-dev** [comment]
Sorry for bothering you, but could you please try again? @pedrobaeza

**Comment by pedrobaeza** [comment]
Rebel modules is conflicting in these cases. It's better to find the compatible way, and at the end that conflict will happen in your DBs:

/ocabot merge nobump

**Comment by francesco-ooops** [comment]
Thanks a lot for the support @pedrobaeza @MiquelRForgeFlow !

**Pull request merged**